packages/core/src/transport/failedSendBeacon.ts

```diff
@@ -0,0 +1,38 @@
+import { isExperimentalFeatureEnabled } from '../domain/configuration'
+import { addMonitoringMessage, monitor } from '../domain/internalMonitoring'
+
+// replaced at build time
+declare const __BUILD_ENV__SDK_VERSION__: string
+
+const FAILED_SEND_BEACON_FLUSH_INTERVAL = 2000
+export const LOCAL_STORAGE_KEY = 'datadog-browser-sdk-failed-send-beacon'
+
+export function startFlushFailedSendBeacons() {
+  if (!isExperimentalFeatureEnabled('lower-batch-size')) return
+
+  setInterval(monitor(flushFailedSendBeacon), FAILED_SEND_BEACON_FLUSH_INTERVAL)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏it is a shame to always write/read the local storage while we only need it for unload/init. Moreover, a flush could also happen during the unload phase, no?
What about only writing in local storage at unload and only reading from it at init?

Reply:
Done